packages/graphql_flutter/test/widgets/query_test.dart

```diff
@@ -370,5 +377,57 @@ void main() {
       await tester.pump();
       verifyNoMoreInteractions(mockHttpClient);
     });
+
+    testWidgets('invokes onComplete callback when completed',
```

Comment:
Can we add a few tests to check that the callbacks are not called too many times?

Reply:
+1